main.py

```diff
@@ -109,8 +115,17 @@ def _request_flow_list(self, switch):
                                                              switch)
             xid_ports = of_core_v0x04_utils.request_port_stats(self.controller,
                                                                switch)
-            self._multipart_replies_xids[switch.id] = {'flows': xid_flows,
-                                                       'ports': xid_ports}
+            self._multipart_replies_xids[switch.id] = {
+                                                        'flows': xid_flows,
+                                                        'ports': xid_ports
+                                                      }
+            if switch.features.capabilities.value & \
```

Comment:
@gretelliz, for unit test completeness can you cover a case when trying to call `__request_stats` but the capabilities `Capabilities.OFPC_TABLE_STATS` isn't set on switch.features.capabilities? and similarly, when it set it does. The PR looks really great and concise, but let's ensure this requirement is also covered with unit test.

Reply:
done!